Addgene_parser.py

```diff
@@ -0,0 +1,389 @@
+        except AttributeError:
+            resistance = None
+
+        # getting Growth Temperature
+        try:
+            growth_t = self.doc.find(string='Growth Temperature').find_parent('li', class_='field').text.split()[2::]
+        except AttributeError:
+            growth_t = None
+
+        # getting Growth Strain(s)
+        try:
+            growth_strain = self.doc.find(string='Growth Strain(s)').find_parent('li', class_='field').text.split()[2::]
+        except AttributeError:
+            growth_strain = None
+
+        # getting Growth instructions
+        try:
+            growth_instructions = ' '.join(
+                self.doc.find(string='Growth instructions').find_parent('li', class_='field').text.split()[2::])
+        except AttributeError:
+            growth_instructions = None
+
+        # getting Copy number
+        try:
+            copy_num = ' '.join(self.doc.find(string='Copy number').find_parent('li', class_='field').text.split()[2::])
+        except AttributeError:
+            copy_num = None
+
+        # getting Gene/Insert name
+        try:
+            gene_insert = ' '.join(
+                self.doc.find(string='Gene/Insert name').find_parent('li', class_='field').text.split()[2::])
+        except AttributeError:
+            gene_insert = None
+
+        # getting Total vector size (bp)
+        try:
+            size = int(' '.join(
+                self.doc.find(string='Total vector size (bp)').find_parent('li', class_='field').text.split()[4::]))
+        except AttributeError:
+            size = None
+
+        plasmid = Plasmid(name=name, gene_insert=gene_insert, growth_instructions=growth_instructions,
+                          copy_num=copy_num, marker=marker,
+                          growth_strain=growth_strain, resistance=resistance, vector_type=vector_type,
+                          backbone=backbone, id=id,
+                          vendor=self.vendor, url=self.url, growth_t=growth_t, size=size)
+
+        plasmid.to_csv(self.path)
+        plasmid.to_txt(self.path, self.doc_seq)
+
+        PlasmidParser.plasmid_list.append(plasmid)
+        return plasmid
+
+    # only Addgene vendor is implemented yet
+    def get_html(self):
+        if self.vendor == 'addgene':
+            url_sequence = self.url + 'sequences/'
+            result_html = requests.get(self.url)
+            result_seq = requests.get(url_sequence)
+            return BeautifulSoup(result_html.text, 'html.parser'), BeautifulSoup(result_seq.text, 'html.parser')
+
+
+def main():
+    """A test function that shows how PlasmidParser works"""
+    if __name__ == '__main__':
+        id_list = [42888, 42876, 26248, 186478, 22222]
+        PlasmidParser(id=id_list)
+        plasmids_for_test = {}
+        for plasmid in PlasmidParser.plasmid_list:
+            plasmids_for_test.update({plasmid.name: plasmid})
+        for k, v in plasmids_for_test.items():
+            print(f"{k}: {v}")
+
+
+main()
+
+"""
```

Comment:
Before that part there is the same code above. I think the code below is a proper one based on the `main()` part?